src/compiler/scala/tools/nsc/backend/opt/ConstantOptimization.scala

```diff
@@ -28,8 +28,8 @@ abstract class ConstantOptimization extends SubComponent {
   import global._
   import icodes._
   import icodes.opcodes._
-  import scala.tools.nsc.backend.icode.Primitives._
-  import scala.tools.nsc.backend.icode.Opcodes._
+  import scala.tools.nsc.backend.jvm.Primitives._
```

Comment:
I really really would like to avoid that, it's very weird, it may break stuff, it's evil. Also I don't see what difference it would make, you still need to split things into different files with different names.